project2-beat-that/script.js

```diff
@@ -1,4 +1,266 @@
+    return `Based on the dice you've rolled, the sum of your score is ${player2Score}. <br><br> Click the buzzer for the next instruction.`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  input = Number(input);
+  // var gameMode;
+  if (gameMode === "player1diceroll") {
+    dicesRolled = get2DiceRolls();
+    gameMode = "player1diceorder";
+    return `Welcome to the game, player ${currentPlayer}. <br><br>
+    You rolled 2 dices. <br>
+    Dice 1: ${dicesRolled[0]}. <br><br>
+    Dice 2: ${dicesRolled[1]}. <br><br>
+    Now, ENTER either 1 or 2 BELOW and click the blue buzzer ABOVE to define the order of your dice roll. This program will then show you your score.
+    `;
+  }
+
+  if (gameMode === "player1diceorder") {
+    if (input != 1 && input != 2) {
+      return `Please only ENTER the number 1 or 2 BELOW. Then, click on the buzzer to check your score.`;
+    }
+    gameMode = "player2turn";
+    currentPlayer = 2;
+    //show return current number player 1 chose.
+    return getPlayer1Score(input);
+  }
+  //reflect instruction to Player 2.
+  if (gameMode === "player2turn") {
+    gameMode = "player2diceroll";
+    return `Okay, now's Player 2's turn. <br><br>
+    Please CLICK blue buzzer to start the game. `;
+  }
+
+  if (gameMode === "player2diceroll") {
+    dicesRolled = get2DiceRolls();
+    gameMode = "player2diceorder";
+    return `Welcome to the game, player ${currentPlayer}. <br><br>
+    You rolled 2 dices. <br>
+    Dice 1: ${dicesRolled[0]}. <br><br>
+    Dice 2: ${dicesRolled[1]}. <br><br>
+    Now, ENTER either 1 or 2 BELOW and click the blue buzzer ABOVE to define the order of your dice roll. This program will then show you your score.
+    `;
+  }
+
+  if (gameMode === "player2diceorder") {
+    if (input != 1 && input != 2) {
+      return `Please only enter the number 1 or 2 BELOW. Then, click on the buzzer to check your score.`;
+    }
+    gameMode = "getreadyforresult";
+    return getPlayer2Score(input);
+  }
+
+  if (gameMode === "getreadyforresult") {
+    gameMode = "getresult";
+    return `Now that both players have played,
+    please click blue buzzer to check who has won. `;
+  }
+
+  if (gameMode === "getresult") {
+    gameMode = "player1diceroll";
+    currentPlayer = 1;
+    if (player1Score > player2Score) {
+      return `Player 1 wins! <br><br>
+      Player 1 score: ${player1Score} <br><br>
+      Player 2 score: ${player2Score}`;
+    }
+    if (player2Score > player1Score) {
+      return `Player 2 wins! <br><br>
+      Player 2 score: ${player2Score} <br><br>
+      Player 1 score: ${player1Score}`;
+    }
+    return `it's a draw! <br><br>
+    Player 1 score: ${player1Score} <br><br>
+    Player 2 score: ${player2Score}`;
+  }
 };
+
+/* need help: In this version, I tried incorporating player 1 and 2 playerScore logic into ONE function (getPlayerScore), but why is player 1 score not shown in results? */
```

Comment:
perhaps you can show me in class or after class what happens when this function runs but i'll leave some comments underneath too